NAS2D/StateManager.cpp

```diff
@@ -35,11 +35,7 @@ StateManager::StateManager() :
  */
 StateManager::~StateManager()
 {
-	if (mActiveState)
-	{
-		Utility<EventHandler>::get().disconnectAll();
-		delete mActiveState;
-	}
+	Utility<EventHandler>::get().disconnectAll();
```

Comment:
I was under the impression that the if guard was to prevent double-deleting `mActiveState` because it is never set to `nullptr` during `setState`. In fact, the only time it was ever `nullptr` was during the constructor but at that point, constructing, then immediately destructing StateManager without doing anything else will call delete on a `nullptr` which is a well-defined no-op.

The use of `unique_ptr` mitigates the use of the guard and since it was being deleted *after* the call to `disconnectAll`, you can leverage the fact that the `unqiue_ptr` will automatically be destructed after the destructor finishes. Destroying a default/unassigned `unique_ptr` is the same as calling deleting on a `nullptr`. Which is what would have happened anyway.